src/IW/Db/Issue.hs

```diff
@@ -35,3 +38,17 @@ getIssuesByLabel label = query [sql|
     FROM issues 
     WHERE ? = ANY (labels)
 |] (Only label)
+
+-- | Inserts a single issue into the database
+insertIssue :: (WithDb env m) => Issue -> m ()
+insertIssue issue = execute [sql|
+    INSERT INTO issues (number, title, body, url, owner, repo_name, labels)
+    VALUES (?, ?, ?, ?, ?, ?, ?)
+|] issue
+
+-- | Insert a list of issues into the database
+insertIssues :: (WithDb env m) => [Issue] -> m ()
+insertIssues issues = executeMany [sql|
+    INSERT INTO issues (number, title, body, url, owner, repo_name, labels)
+    VALUES (?, ?, ?, ?, ?, ?, ?)
```

Comment:
There is a potential problem with this function. You can insert issue which doesn't have the corresponding repository, so DB will throw you an error about foreign key constraint violation. This can easily happen if your async work will have the following steps:

1. Sync all repos.
2. Sync all issues.

Repositories and users can be renamed before you fetch all issues, new repos and new issues can appear before you fetched all issues but synced all repos.

Could you please try to play with SQL in REPL using INSERT with `ON CONFLICT DO NOTHING` clause to verify that if FK violates we can just ignore such issues?

* https://www.postgresql.org/docs/9.5/sql-insert.html

If it doesn't work, I propose you to play with this function in GHCi REPL to check the behavior during error: will it stop on first error or will it insert everything anyways? If we have undesired behavior, I propse to create a question on SO about this problem: How to ignore insert in case of foreign key violation?

Reply:
Understood. I will try to test this out and see what happens.

Reply:
Oh, one more thing I forgot. We actually need UPSERT query here. We need UPDATE when the object exists and INSERT otherwise. Check also how this is usually done in PostgreSQL.

Reply:
UPSERT? Okay I will research this

Reply:
Okay so I've done a bit of looking. It seems like there isn't a simple way to ignore the foreign key constraint and insert the rest of the issues. `ON CONFLICT DO NOTHING` still ignores the rest of the issues. I'm still using `ON CONFLICT DO UPDATE ....` to perform `UPSERT`s. This is no problem. The most common solution I can find for ignoring the foreign key constraint is first inserting the issues into something called a staging table, filtering the issues that don't have correct foreign keys, and then inserting into the actual issues table. I'm still doing more research.

Reply:
Apparently, you can also use triggers. Not sure which is better.  https://stackoverflow.com/questions/39827797/postgres-ignore-insert-error-and-carry-ony

Reply:
These also seem to be valid solutions:
https://stackoverflow.com/questions/36038751/can-insert-on-conflict-be-used-for-foreign-key-violations/36039580#36039580

https://stackoverflow.com/questions/45228853/how-to-stop-the-insert-or-update-on-table-violates-foreign-key-constraint?rq=1